pkg/client/custom_writer.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"context"
+	monv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/prometheus"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientv1 "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CustomWriter provides capability to identify client at runtime
+type customWriter struct {
+	Client *Client
+}
+
+// CustomWriterC provide writer reference for given client
+func CustomWriterC(cli *Client) clientv1.Writer {
+	return &customWriter{
+		Client: cli,
+	}
+}
+
+// Create saves the object obj in the Kubernetes cluster.
+func (w *customWriter) Create(ctx context.Context, obj runtime.Object, opts ...clientv1.CreateOption) error {
```

Comment:
Same here for all these functions. Again `resources` is doing the same thing.

Reply:
I had removed custom_reader/custer_writer files